backend/src/main/java/de/neuefische/ffmjava221/teamprojekt/backend/weather/WeatherService.java

```diff
@@ -0,0 +1,37 @@
+package de.neuefische.ffmjava221.teamprojekt.backend.weather;
+
+import org.springframework.beans.factory.annotation.Value;
+
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.util.Objects;
+
+
+@Service
+public class WeatherService {
+    private final WebClient webClient;
+
+    public WeatherService(@Value("${baseUrl}") String baseUrl) {
+        this.webClient = WebClient.create(baseUrl);
+    }
+
+    public WeatherData fetchWeather(String date, int hour) throws IllegalArgumentException, WeatherResponseIsNullException {
+        if (hour < 0 || hour > 24) {
+            throw new IllegalArgumentException("Hour must be between 0 and 24");
+        }
+        WeatherForecastResponse weatherResponse = Objects.requireNonNull(webClient
+                        .get()
+                        .uri("?wmo_station_id=10637&date=" + date)
+                        .retrieve()
+                        .toEntity(WeatherForecastResponse.class)
+                        .block())
+                        .getBody();
```

Comment:
Try to seperate the response from the body. 

` ResponseEntity<WeatherForecastResponse> weatherResponse = (webClient
                        .get()
                        .uri("?wmo_station_id=10637&date=" + date)
                        .retrieve()
                        .toEntity(WeatherForecastResponse.class)
                        .block());`